client/src/components/NavigationBar.tsx

```diff
@@ -6,6 +6,11 @@ import { NAV_CONFIG } from "../App.tsx";
 function NavigationBar() {
   const location = useLocation().pathname;
   const navigate = useNavigate();
+
+  const handleProfileClick = () => {
+    navigate(NAV_CONFIG.profile.href);
```

Comment:
Can't we just use the `Link` element?

Reply:
Because it does a server side redirect, we want client side redirects